### PR TITLE
Split rewards from overview query

### LINKED ISSUE #3806: Split rewards from other queries
**Is your feature request related to a problem? Please describe.**
Rewards queries are slow in Cosmos and Polkadot.

**Describe the solution you'd like**

Split rewards from other queries are implemented in API, so let's split them also in the frontend, that way we don't block the UI and improve our UX.

**Additional context**
https://github.com/luniehq/lunie-api/issues/531

### PR DESCRIPTION
**Pull request by mariopino**
Closes #3806

**Description:**

That's it, by now uses a computed field to calculate `totalRewards` until we implement BalancesV2 that includes that property.

**RELATED:** https://github.com/luniehq/lunie-api/pull/543

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/components/common/TmBalance.vue [hunk 1]
```diff
@@ -346,14 +344,23 @@ export default {
...
       } else return {}
+    },
+    totalRewards() {
+      if (this.rewards && this.rewards.length > 0) {
+        return this.rewards
+          .filter(({ denom }) => denom === this.stakingDenom)
+          .reduce((sum, { amount }) => parseFloat(amount) + sum, 0)
+          .toFixed(6)
+      }
+      return 0
```
Comment (Bitcoinera) [inline src/components/common/TmBalance.vue:363]: Looking good!

File: src/components/common/TmBalance.vue [hunk 2]
```diff
@@ -346,14 +345,28 @@ export default {
...
+          .filter(({ denom }) => denom === this.stakingDenom)
+          .reduce((sum, { amount }) => parseFloat(amount) + sum, 0)
+          .toFixed(6)
+        if (!this.rewardsSentToGA) {
+          this.sendRewards(totalRewards)
+        }
+
+        return totalRewards
+      }
+      return 0
```
Comment (faboweb) [inline src/components/common/TmBalance.vue:369]:
```suggestion
      return this.totalRewardsPerDenom[this.stakingDenom] || 0
```

Comment (faboweb) [reply src/components/common/TmBalance.vue:369]: you can move
```
if (!this.rewardsSentToGA) {
          this.sendRewards(totalRewards)
        }
```
to a watcher or inside `totalRewardsPerDenom`

Comment (Bitcoinera) [reply src/components/common/TmBalance.vue:369]: Done

File: src/components/common/TmBalance.vue [hunk 3]
```diff
@@ -346,6 +346,9 @@ export default {
     },
     totalRewardsPerDenom() {
       if (this.rewards && this.rewards.length > 0) {
+        if (!this.rewardsSentToGA) {
+          this.sendRewards(this.totalRewards)
```
Comment (faboweb) [inline src/components/common/TmBalance.vue:350]: total rewards is not available yet at this point, is it?

Comment (faboweb) [reply src/components/common/TmBalance.vue:350]: @Bitcoinera

Comment (Bitcoinera) [reply src/components/common/TmBalance.vue:350]: yes, I checked. But I can add it to the check for more security

Comment (faboweb) [reply src/components/common/TmBalance.vue:350]: No, I mean `totalRewards` depends on `totalRewardsPerDenom` so at this point the value hasn't updated yet. As I recommended before: Create a `watch` for `totalRewards` and send it there. But we will also change this to send all the analytics inside the API.

Comment (Bitcoinera) [reply src/components/common/TmBalance.vue:350]: xDDD you suggested both things as possibilities XD

And nope totalRewards doesn't depend on totalRewardsPerDenom. Look at the code

Comment (faboweb) [reply src/components/common/TmBalance.vue:350]: with this it would https://github.com/luniehq/lunie/pull/3818#discussion_r401627601

Comment (faboweb) [reply src/components/common/TmBalance.vue:350]: I pushed a proposal

Comment (mariopino) [reply src/components/common/TmBalance.vue]: What's missing to merge here?

File: src/components/common/TmBalance.vue [hunk 4]
```diff
@@ -238,10 +238,9 @@ export default {
   },
   data() {
     return {
-      overview: {
-        rewards: []
-      },
+      overview: {},
       sentToGA: false,
+      rewardsSentToGA: false,
```
Comment (jbibla) [inline src/components/common/TmBalance.vue:243]: what's this?

File: src/components/common/TmBalance.vue [hunk 5]
```diff
@@ -538,6 +552,7 @@ export default {
           /* istanbul ignore next */
           this.$apollo.queries.overview.refetch()
           this.$apollo.queries.balances.refetch()
+          this.$apollo.queries.rewards.refetch()
```
Comment (jbibla) [inline src/components/common/TmBalance.vue:555]: are we batching these requests @faboweb ?

File: src/components/common/TmBalance.vue (hunk 4 above)
Comment (Bitcoinera) [reply src/components/common/TmBalance.vue:243]: a checker to see if `totalRewards` were already sent to GA

File: src/components/common/TmBalance.vue (hunk 5 above)
Comment (faboweb) [reply src/components/common/TmBalance.vue:555]: I think so, would need to check myself

**Pull request merged**